packages/blocks/src/api/factory.js

```diff
@@ -117,26 +118,41 @@ const isPossibleTransformForSource = ( transform, direction, blocks ) => {
 		return false;
 	}
 
-	// If multiple blocks are selected, only multi block transforms are allowed.
+	// If multiple blocks are selected, only multi block transforms
+	// or wildcard transforms are allowed.
 	const isMultiBlock = blocks.length > 1;
-	const isValidForMultiBlocks = ! isMultiBlock || transform.isMultiBlock;
+	const firstBlockName = first( blocks ).name;
+	const isValidForMultiBlocks = isWildcardBlockTransform( transform ) || ! isMultiBlock || transform.isMultiBlock;
 	if ( ! isValidForMultiBlocks ) {
 		return false;
 	}
 
+	// Check non-wildcard transforms to ensure that transform is valid
+	// for a block selection of multiple blocks of different types
+	if ( ! isWildcardBlockTransform( transform ) && ! every( blocks, { name: firstBlockName } ) ) {
+		return false;
+	}
+
 	// Only consider 'block' type transforms as valid.
 	const isBlockType = transform.type === 'block';
 	if ( ! isBlockType ) {
 		return false;
 	}
 
-	// Check if the transform's block name matches the source block only if this is a transform 'from'.
+	// Check if the transform's block name matches the source block (or is a wildcard)
+	// only if this is a transform 'from'.
 	const sourceBlock = first( blocks );
-	const hasMatchingName = direction !== 'from' || transform.blocks.indexOf( sourceBlock.name ) !== -1;
+	const hasMatchingName = direction !== 'from' || transform.blocks.indexOf( sourceBlock.name ) !== -1 || isWildcardBlockTransform( transform );
 	if ( ! hasMatchingName ) {
 		return false;
 	}
 
+	// Don't allow single 'core/group' blocks to be transformed into
```

Comment:
In pursuit of avoiding block-specific behaviors here: Is this more about preventing a block from being transformed into itself, which could apply to any block type and not specifically the group block?

Reply:
I guess this could be applied generically. Trying to think of any cases where transform into a single block of the same type might be valid...

Reply:
See https://wordpress.slack.com/archives/C02QB2JS7/p1559722654012000

Reply:
Issue raised https://github.com/WordPress/gutenberg/issues/15996